### PR TITLE
201 update dependencies

### LINKED ISSUE #194: Installation Error:  Building wheel for scikit-learn (pyproject.toml) did not run successfully.
Dear all, 

I am trying to install ojd-daps-skills via the command

`pip install ojd-daps-skills`

however, the installation stopped due to errors in building wheels for scikit-learn.  The error message is rather long, see here: 
[error_log.txt](https://github.com/nestauk/ojd_daps_skills/files/12192474/error_log.txt)

I extracted some as follows: 

... 
Building wheels for collected packages: scikit-learn
  Building wheel for scikit-learn (pyproject.toml) ...
error: subprocess-exited-with-error

  × Building wheel for scikit-learn (pyproject.toml) did not run successfully.
... 

I have tried upgrading pip, cython, setuptools but the error still persisted.  
The environment where the error occured: 
Python 3.9.13, 
Windows 10, 
Cython version 3.0.0, 
pip 23.2.1

I also tried installing it in Google Colab and the problem persists. 

Most grateful for any helps.  Thanks !

### LINKED ISSUE #199: Upcoming Maintenance 🌟
**EDIT:** We've rolled maintence into an upcoming project so new features/refactoring will not be added until early next year. However, the new release should address bugs. 

Hi ojd_daps_skills stargazers! 

👂 You've been tagged as we'd love to hear from you! 

🐛 We will have resource to address bugs, add new features and re-release the library next month. There are immediate issues we will deal with (aws cli problems etc.) and new methods we want to release. 

We'd also like to hear if there are any particular requests you have. Feel free to add to this issue and we will add it to our maintenance backlog. 

Thanks! 

@JoseMoreville 
@hengkiardo 
@mpaulacaldas 
@jibbolo 
@ppseguel 
@Bhanu-py 
@jamesweiner 
@ephraimberkovitch 
@Hao-Z-hang 
@smrutil 
@luisgasco 
@bouacheya 
@Timliuzhaolu 
@mhaeussermann 
@francesco-trentini 
@n3ssuno 
@kstathou 
@FabienPetitEconomics 
@nikolasjdawson 
@nguaman 
@parraguezr

### LINKED ISSUE #200: [Suggestion] Switch the need to have the aws cli to using the boto3 python library.
Using an unsigned boto3 client you can download the required zip file without the need to have the aws cli installed.
Following code below works and is completely python based ( makes the project more multi OS friendly aswell ).

```python
import os
import boto3
from botocore.exceptions import ClientError
from botocore import UNSIGNED
from botocore.config import Config
import zipfile

PUBLIC_DATA_FOLDER_NAME = 'ojd_daps_skills_data'
PROJECT_DIR = os.getcwd()

def download():

    aws_region = 'eu-west-1'
    s3 = boto3.client('s3', region_name=aws_region, config=Config(signature_version=UNSIGNED))

    bucket_name = 'open-jobs-indicators'
    key = f'escoe_extension/{PUBLIC_DATA_FOLDER_NAME}.zip'
    public_data_dir = os.path.join(PROJECT_DIR, PUBLIC_DATA_FOLDER_NAME)

    try:
        s3.download_file(bucket_name, key, f'ojd_daps_skills_data.zip')

        with zipfile.ZipFile(f'{public_data_dir}.zip', 'r') as zip_ref:
            zip_ref.extractall(PROJECT_DIR)

        os.remove(f'{public_data_dir}.zip')

    except ClientError as e:
        print(f"Error: {e}")

download()
```

### LINKED ISSUE #201: Cython, Numpy, PyYAML, awscli dependencies updates
This issue addresses necessary dependency updates due to regressions from Cython 3 release. This issue will be linked to a Work-in-progress PR and branch, to be merged to a wider maintenance PR and/ or changelog. The goal is to make ojd_daps_skills `pip install`able and `make install`able again.

To-do are as follows:
- [x] Remove awscli from requirements_dev.txt
- [x] Update scikit-learn, numpy, and libraries depending on these (sentence-transformers, transformers)
- [ ] Expand and test compatibility with Windows
- [ ] Test above versions for conflicts with model training and other pipelines.

### PR DESCRIPTION
**Pull request by sqr00t**
- Removes awscli from requirements_dev.txt
- Updates following dependencies:
  - numpy 1.22.4 -> 1.24.4
  - scipy 1.8.1 -> 1.10.1
  - scikit-learn 0.23.2 -> 1.3.1
- Adapts @Aiden-RC 's snippet for download() function using boto3
- Adds pip dependency caching for "Unit Tests" workflow
- Adds windows-latest support to "Unit Tests" Actions workflow
- Utilise Path objects in setup.py
- Fixes cross-platform version tagging bug by:
  - Adding setuptools_scm to setup.py
  - Build config to a pyproject.toml file
- Removes manually downloading base SpaCy model

---

Thanks for contributing to Nesta's Skills Extractor Library 🙏!

If you have suggested changes to _code_ anywhere outside of the ExtractSkills class, please consult the checklist below.

Checklist ✔️🐍:

- [x] I have refactored my code out from `notebooks/` (no notebook code involved)
- [x] I have checked the code runs
- [x] I have tested the code
- [x] I have run `pre-commit` and addressed any issues not automatically fixed
- [x] I have merged any new changes from `dev`
- [x] I have documented the code
  - [x] Major functions have docstrings
  - [x] Appropriate information has been added to `README`s
- [x] I have explained the feature in this PR or (better) in `output/reports/`
- [x] I have requested a code review

If you have suggested changes to _documentation_ (and/or the ExtractSkills class), please **ALSO** consult the checklist below.

Documentation Checklist ✔️📚:

- [x] I have run `make html` in `docs`
- [x] I have manually reviewed the `docs/build/*.html` files locally to ensure they have formatted correctly
- [x] I have pushed both relevant files AND their corresponding `docs/build/*.html` files

### CODE REVIEW

File: ojd_daps_skills/getters/download_public_data.py [hunk 1]
```diff
@@ -14,5 +13,43 @@ def download():
...
+
+import os
+import boto3
+from botocore.exceptions import ClientError
+from botocore import UNSIGNED
+from botocore.config import Config
+from zipfile import ZipFile
+
+
+def download():
```
Comment (Aiden-RC) [inline ojd_daps_skills/getters/download_public_data.py:10]: i'm unsure if you've realised but you've defined and called both the aws cli command aswell as using boto to download the data?
You've also overwritten the same function

**Review by Aiden-RC — CHANGES_REQUESTED** [review]
Unsure why download function has been overwritten / called twice?
Potential oversight?

File: ojd_daps_skills/getters/download_public_data.py (hunk 1 above)
Comment (sqr00t) [reply ojd_daps_skills/getters/download_public_data.py:10]: Brilliant spot! I had a long day. Thank you 🤣

**Comment by Aiden-RC** [comment]
Have tested, works perfectly for me.
As was running on windows had to make a few changes to get it to run ( that i had made previously but had to do a few other changes also to make it run on windows correctly.
New PR made for this at https://github.com/nestauk/ojd_daps_skills/pull/204
contains this PR changes aswell as windows related changes and windows pytest actions.

I would recommend removing aws cli section of readme like i did on my PR if mine doesn't get merged

File: README.md [hunk 2]
```diff
@@ -35,10 +35,6 @@ You will also need to download [spaCy's](https://spacy.io/models/en) `en_core_we
 python -m spacy download en_core_web_sm
 ```
 
-### AWS CLI
```
Comment (Aiden-RC) [inline README.md:39]: This conflicts with latest changes on the 'dev' branch.

**Comment by sqr00t** [comment]
I've had a deeper look at the version tagging implementation.
Automatic versioning now should work across Windows and Mac:
- Added using setuptools_scm for this feature
- Added some build config in a new pyproject.toml file

Further work that may be needed:
- Major/ Minor releases may need to have it's own branch created manually, as this automatically updates the fix/hotfix version. I'm having another look at [this](https://riggaroo.dev/using-github-actions-to-automate-our-release-process/)

**Comment by india-kerle** [comment]
great - i was able to create a new env, install dependencies and use the libraries core functionality with no errors. Looks like all tests on ubuntu and windows passed as well.

**Review by india-kerle — APPROVED** [review]

**Pull request merged**